File: NzolaWebAPI/Dtos/Notificacao/NotificacaoDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using NzolaWebAPI.DTOs.Utilizador;
using NzolaWebAPI.Models.Enums;


namespace NzolaWebAPI.DTOs.Notificacao
{
    public class NotificacaoDto
    {
        
        public int Id {get; set;}
        public int UtilizadorId {get; set;}
        public TipoNotificacao Tipo {get; set;}
        public UtilizadorSimplificadoDto? UtilizadorNotificacao { get; set; }

        public int OrigemId { get; set; }
        public UtilizadorSimplificadoDto? UtilizadorResponsavel { get; set; }
        public string? Mensagem {get; set;}
        public bool Lida {get; set;} 
        public DateTime CriadoEm {get; set;} 
    }
    
}

File: NzolaSPA/src/app/dtos/notificacao/notificacao.dto.ts
import { UtilizadorSimplificadoDto } from "../utilizador/utilizadorfeed/utilizador.dto";

export class NotificacaoDto {
  id: number = 0;
  utilizadorId: number = 0;
  tipo: string = '';
  origemId: number = 0;
  mensagem: string = '';
  utilizadorNotificacao: UtilizadorSimplificadoDto = new UtilizadorSimplificadoDto();
  utilizadorResponsavel: UtilizadorSimplificadoDto = new UtilizadorSimplificadoDto();
  lida: boolean = false;
  criadoEm : string = '';
}

export class NovaNotificacaoDto {
  utilizadorId: number = 0;
  tipo: number = 0;
  origemId: number = 0;
  mensagem: string = '';
}
